kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/Annotations.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.ObjectMetaFluent;
+import io.fabric8.kubernetes.client.utils.KubernetesResourceUtil;
+
+import io.kroxylicious.proxy.tag.VisibleForTesting;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Used to read/write annotations to/from Kubernetes resources and associated builders. This
+ * class aims to encapsulate annotation logic so that we can make breaking changes to the annotation
+ * keys or data, without changing client interfaces.
+ */
+public class Annotations {
```

Comment:
I'm not 100% sold that this is the right pattern for the long term but I do think its a reasonable base and is pushing us towards the right kind of pattern in hiding the details.

Reply:
yeah, can revisit later, it's a bit funky having all the annotations defined in one spot when some are for purpose A and some for purpose B. It should be arranged so that there's this component with the common read/writing logic, and some way of plugging in the details for each annotation type.